config/config.go

```diff
@@ -0,0 +1,16 @@
+package config
+
+import (
+	_ "embed"
+
+	"gopkg.in/yaml.v3"
+)
+
+//go:embed config.yaml
+var config string
+
+func GetConfig() map[string]interface{} {
```

Comment:
Using map is a valid approach here. Maybe you can consider introducing a `Config` structure that will contains all expected configurable fields. You can unmarshal to this struct only once and reuse it across the project.

Benefits:
- no need in parsing map and accessing through keys;
- no need to type cast every like (`v, ok := val.(string)`);
- easy to understand what we deal with.

```go
type Config struct {
	ProjectID  string `yaml:"project_id"`
	Region     string `yaml:"region"`
	OtherValue int32  `yaml:"other_value"`
}

func (c Config) Validate() error {
	// Easy to validate.
	if c.ProjectID != "us-central1" {
		return errors.New("wrong region")
	}
	return nil
}

func testFunc(cfg Config) {
	// Easy to use without a type cast.
	projectID := cfg.ProjectID
	// ...
}
```

Reply:
I completely agree that this should be a struct, I just didn't take the time to do it yet since I only have one field in there right now. Totally agree, though

Reply:
So after looking at this and how it loads into viper, I may just leave this as a map - viper has a [nice and easy to use function](https://pkg.go.dev/github.com/spf13/viper#MergeConfigMap) that loads a map into its config. If this gets too complicated, I'm open to moving to a struct, but for now I think I'll keep it simple